File: src/test/java/example/example/pages/FacebookLoginPage.java
package example.example.pages;

import com.microsoft.playwright.Page;

public class FacebookLoginPage extends BasePage {

	public FacebookLoginPage(Page page) {
		super(page);
	}

	public FacebookLoginPage enterEmail(String email) {
		page.locator("#email").fill(email);
		return this;
	}

	public FacebookLoginPage enterPassword(String password) {
		page.locator("#pass").fill(password);
		return this;
	}

	public void clickSignIn() {
		page.locator("[name='login']").click();
	}
}

File: src/test/python/example/example/pages/facebook_login_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.remote.webdriver import WebDriver

from example.example.pages.base_page import BasePage


class FacebookLoginPage(BasePage):

    EMAIL_INPUT = (By.ID, "email")
    PASSWORD_INPUT = (By.ID, "pass")

    def __init__(self, driver: WebDriver) -> None:
        super().__init__(driver)

    def enter_email(self, email: str) -> "FacebookLoginPage":
        email_input = self.driver.find_element(*self.EMAIL_INPUT)
        email_input.send_keys(email)
        return self

    def enter_password(self, password: str) -> "FacebookLoginPage":
        pass_input = self.driver.find_element(*self.PASSWORD_INPUT)
        pass_input.send_keys(password)
        return self

    def click_sign_in(self) -> None:
        pass_input = self.driver.find_element(*self.PASSWORD_INPUT)
        pass_input.submit()
